static/js/stream_data.js

```diff
@@ -726,6 +726,38 @@ exports.create_sub_from_server_data = function (stream_name, attrs) {
     return sub;
 };
 
+exports.get_unmatched_streams_for_notification_settings = function () {
+    const stream_notification_settings = get_stream_notification_settings();
+    const subscribed_rows = exports.subscribed_subs();
+    function by_name(a, b) {
+        return util.strcmp(a.name, b.name);
+    }
+    subscribed_rows.sort(by_name);
+
+    const notification_settings = [];
+    for (const row of subscribed_rows) {
+        const settings_values = {};
+        let is_stream_unmatched = false;
+        for (const notification_name of stream_notification_settings) {
+            const prepend = notification_name === 'wildcard_mentions_notify' ? "" : "enable_stream_";
+            const global_settings = page_params[prepend + notification_name];
+            const stream_settings = exports.receives_notifications(row.name, notification_name);
+
+            const not_matched = stream_settings !== global_settings;
+            settings_values[notification_name] = not_matched;
+            is_stream_unmatched = is_stream_unmatched || not_matched;
```

Comment:
I'm confused, isn't `is_stream_unmatched` always `false` at this point? Couldn't we just be using `not_matched` for the check below and do away with `is_stream_unmatched`? But maybe I'm reading this wrong.

Reply:
Added `is_stream_unmatched` as we want to find whether the value of `not_matched` is true during any iteration.
I could change it to 
```
if (not_matched)
    is_stream_unmatched = true;
```
if helps in improving readability?

Reply:
Hmm not sure. I was referring to the fact that you reset `is_stream_unmatched` to `false` every time before entering this inner loop.  So all your doing is basically assigning the value of `not_matched` to `is_stream_unmatched`?

Reply:
Ohh, sorry. Okay I see what you mean.